packages/frontEnd/src/views/categoryDetails/CategoryDetails.tsx

```diff
@@ -20,18 +19,22 @@ export const CategoryDetails = () => {
   const [currentPage, setCurrentPage] = useState(1);
   const [products, setProducts] = useState<null | ProductProps[]>(null);
   const [totalPages, setTotalPages] = useState(0);
-  const [category, setCategory] = useState("");
-  const [rate, setRate] = useState("");
-  const [city, setCity] = useState("");
   const [searchParams, setSearchParams] = useSearchParams();
 
   useEffect(() => {
+    let params = {};
+
+    // @ts-ignore
+    for (const [key, value] of searchParams.entries()) {
+      // @ts-ignore
+      params[key] = value;
+    }
+
```

Comment:
```suggestion
  type Obj = {
    [key: string]: string;
  };

  useEffect(() => {
    let params: Obj = {};

    for (const [key, value] of searchParams.entries()) {
      params[key] = value;
    }
```
or
 ```suggestion
  type Obj = Record<string,string>

  useEffect(() => {
    let params: Obj = {};

    for (const [key, value] of searchParams.entries()) {
      params[key] = value;
    }
```
To make it work, you need to change target to "es2015" in tsconfig.json